### PR TITLE
Add utility class for performing an action only if at least N milliseconds have passed since the last time.

### PR DESCRIPTION
**Pull request by slipher**
I want to use this to log warnings for some conditions which may occur every frame, without causing massive spam.

### CODE REVIEW

**Comment by DolceTriade** [comment]
It would be helpful if you could commit this with how you use it.

File: src/common/Util.h [hunk 1]
```diff
@@ -181,6 +181,29 @@ template<typename T> class uninitialized {
...
+		int now = Sys_Milliseconds();
+		if (now >= lastTime && now < lastTime + duration) {
+			return false;
+		} else {
+			lastTime = now;
+			return true;
+		}
+	}
+private:
+	int duration;
```
Comment (DolceTriade) [inline src/common/Util.h]: const

File: src/common/Util.h [hunk 2]
```diff
@@ -181,6 +181,29 @@ template<typename T> class uninitialized {
...
+	int lastTime;
```
Comment (DolceTriade) [inline src/common/Util.h]: const

File: src/common/Util.h [hunk 3]
```diff
@@ -181,6 +181,29 @@ template<typename T> class uninitialized {
...
 
+// Use to perform an action only if a minimum time has passed since the last time it was performed.
+// Not thread-safe.
+class MinimumDelay {
+public:
+	// duration: length of delay in milliseconds
+	MinimumDelay(int duration): duration(duration), lastTime(std::numeric_limits<int>::min()) {}
+
+	// Returns true if it is OK to perform the action again.
+	bool check() {
```
Comment (DolceTriade) [inline src/common/Util.h]: const function. Also, I think our style has us capitalizing function names.

File: src/common/Util.h (hunk 1 above)
Comment (slipher) [reply src/common/Util.h]: Done.

File: src/common/Util.h (hunk 3 above)
Comment (slipher) [reply src/common/Util.h]: It is a key part of this method's contract that the internal state is modified.

**Pull request merged**